scene/3d/xr_body_modifier_3d.cpp

```diff
@@ -0,0 +1,415 @@
+		"RightMiddleProximal", // XRBodyTracker::JOINT_RIGHT_MIDDLE_FINGER_PHALANX_PROXIMAL
+		"RightMiddleIntermediate", // XRBodyTracker::JOINT_RIGHT_MIDDLE_FINGER_PHALANX_INTERMEDIATE
+		"RightMiddleDistal", // XRBodyTracker::JOINT_RIGHT_MIDDLE_FINGER_PHALANX_DISTAL
+		"RightMiddleTip", // XRBodyTracker::JOINT_RIGHT_MIDDLE_FINGER_TIP
+		"RightRingMetacarpal", // XRBodyTracker::JOINT_RIGHT_RING_FINGER_METACARPAL
+		"RightRingProximal", // XRBodyTracker::JOINT_RIGHT_RING_FINGER_PHALANX_PROXIMAL
+		"RightRingIntermediate", // XRBodyTracker::JOINT_RIGHT_RING_FINGER_PHALANX_INTERMEDIATE
+		"RightRingDistal", // XRBodyTracker::JOINT_RIGHT_RING_FINGER_PHALANX_DISTAL
+		"RightRingTip", // XRBodyTracker::JOINT_RIGHT_RING_FINGER_TIP
+		"RightLittleMetacarpal", // XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_METACARPAL
+		"RightLittleProximal", // XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_PHALANX_PROXIMAL
+		"RightLittleIntermediate", // XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_PHALANX_INTERMEDIATE
+		"RightLittleDistal", // XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_PHALANX_DISTAL
+		"RightLittleTip", // XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_TIP
+	};
+
+	// reset JIC.
+	for (int i = 0; i < XRBodyTracker::JOINT_MAX; i++) {
+		joints[i].bone = -1;
+		joints[i].parent_joint = -1;
+	}
+
+	Skeleton3D *skeleton = get_skeleton();
+	if (!skeleton) {
+		return;
+	}
+
+	// Find the skeleton-bones associated with each joint.
+	int bones[XRBodyTracker::JOINT_MAX];
+	for (int i = 0; i < XRBodyTracker::JOINT_MAX; i++) {
+		// Skip upper body joints if not enabled.
+		if (!body_update.has_flag(BODY_UPDATE_UPPER_BODY) && i >= XRBodyTracker::JOINT_HIPS && i <= XRBodyTracker::JOINT_RIGHT_LOWER_ARM) {
+			bones[i] = -1;
+			continue;
+		}
+
+		// Skip lower body joints if not enabled.
+		if (!body_update.has_flag(BODY_UPDATE_LOWER_BODY) && i >= XRBodyTracker::JOINT_LEFT_UPPER_LEG && i <= XRBodyTracker::JOINT_RIGHT_TOES) {
+			bones[i] = -1;
+			continue;
+		}
+
+		// Skip hand joints if not enabled.
+		if (!body_update.has_flag(BODY_UPDATE_HANDS) && i >= XRBodyTracker::JOINT_LEFT_HAND && i <= XRBodyTracker::JOINT_RIGHT_PINKY_FINGER_TIP) {
+			bones[i] = -1;
+			continue;
+		}
+
+		// Find the skeleton bone.
+		bones[i] = skeleton->find_bone(bone_names[i]);
+		if (bones[i] == -1) {
+			WARN_PRINT(vformat("Couldn't obtain bone for %s", bone_names[i]));
+		}
+	}
+
+	// If the root bone is not found then use the skeleton root bone.
+	if (bones[XRBodyTracker::JOINT_ROOT] == -1) {
+		bones[XRBodyTracker::JOINT_ROOT] = 0;
+	}
+
+	// Assemble the joint relationship to the available skeleton bones.
+	for (int i = 0; i < XRBodyTracker::JOINT_MAX; i++) {
+		// Get the skeleton bone (skip if not found).
+		const int bone = bones[i];
+		if (bone == -1) {
+			continue;
+		}
+
+		// Find the parent skeleton-bone.
+		const int parent_bone = skeleton->get_bone_parent(bone);
+		if (parent_bone == -1) {
+			// If no parent skeleton-bone exists then drive this relative to palm joint.
```

Comment:
The comment needs to be updated since this is being driven relative to the root joint.

Reply:
Good catch - comment updated!